packages/vats/test/localchain.test.js

```diff
@@ -0,0 +1,156 @@
+import { AmountMath, AssetKind, makeIssuerKit } from '@agoric/ertp';
+import { buildRootObject } from '../src/vat-bank.js';
+import { prepareLocalChainTools } from '../src/localchain.js';
+
+/** @type {import('ava').TestFn<ReturnType<makeTestContext>>} */
+const test = anyTest;
+
+const makeTestContext = async t => {
+  const { fakeVomKit } = reincarnate({ relaxDurabilityRules: false });
+  const provideBaggage = key => {
+    const root = fakeVomKit.cm.provideBaggage();
+    const zone = makeDurableZone(root);
+    return zone.mapStore(`${key} baggage`);
+  };
+  const baggage = provideBaggage('localchainTest');
+  const bankVat = E(buildRootObject)(null, null, baggage);
+
+  const zone = makeDurableZone(baggage);
+
+  /** @type {undefined | ERef<import('../src/types.js').BridgeHandler>} */
+  let bankHandler;
+
+  /** @type {import('../src/types.js').ScopedBridgeManager} */
+  const fakeBridgeManager = zone.exo('fakeBankBridgeManager', undefined, {
+    async fromBridge(obj) {
+      t.is(typeof obj, 'string');
+    },
+    async toBridge(obj) {
+      let ret;
+      switch (obj.type) {
+        case 'VLOCALCHAIN_ALLOCATE_ADDRESS': {
+          ret = 'agoricfoo';
+          break;
+        }
+        case 'VBANK_GET_BALANCE': {
+          const { address, denom, type: _type, ...rest } = obj;
+          t.is(address, 'agoricfoo');
+          t.deepEqual(rest, {});
+          if (denom === 'ubld') {
+            ret = '100000000';
+          } else {
+            t.fail(`unrecognized denomination ${denom}`);
+          }
+          break;
+        }
+
+        default: {
+          t.is(obj, null);
+        }
+      }
+      return ret;
+    },
+    initHandler(newHandler) {
+      bankHandler = newHandler;
+    },
+    setHandler(newHandler) {
+      bankHandler = newHandler;
+    },
+  });
+  const bankManager = await E(bankVat).makeBankManager(fakeBridgeManager);
+
+  const { makeLocalChain } = prepareLocalChainTools(zone.subZone('localchain'));
+
+  const powers = {
+    system: fakeBridgeManager,
+    bankManager,
+  };
+  const localchain = makeLocalChain(powers);
+  const bldIssuerKit = makeIssuerKit(
+    'BLD',
+    AssetKind.NAT,
+    harden({ decimalPlaces: 6 }),
+  );
+
+  const oneHundredBldAmt = AmountMath.make(bldIssuerKit.brand, 100_000_000n);
+  const oneHundredBldPmt = bldIssuerKit.mint.mintPayment(oneHundredBldAmt);
+
+  const beanIssuerKit = makeIssuerKit(
+    'BEAN',
+    AssetKind.NAT,
+    harden({ decimalPlaces: 6 }),
+  );
+  const oneHundredBeanAmt = AmountMath.make(beanIssuerKit.brand, 100_000_000n);
+
+  return {
+    baggage,
+    zone,
+    bankHandler,
+    fakeBridgeManager,
+    bankVat,
+    bankManager,
+    localchain,
+    bldIssuerKit,
+    bldIssuer: bldIssuerKit.issuer,
+    oneHundredBldPmt,
+    oneHundredBldAmt,
+    oneHundredBeanAmt,
+  };
+};
+
+test.beforeEach(t => {
+  t.context = makeTestContext(t);
+});
+
+test('localchain - deposit and withdraw', async t => {
+  const user = async () => {
```

Comment:
Thanks for organizing the test into scopes representing parties / processes.

But... does a user get access to all of `localchain`?

Maybe this should be...

```suggestion
  const anOrchestrationContract = async () => {
```